src/site_adapters/domScrapingBase.ts

```diff
@@ -92,6 +102,20 @@ export function createDomScrapingAdapter(config:ScrapingAdapterConfig):TableAdap
   let scrapedRows: Array<ScrapedRow> = [];
   let sortOrder: SortConfig = null;
   let subscribers: Array<(Table) => void> = [];
+  let scrapedAjaxRows;
+  let scrapedAjaxRowDict = {};
+
+  // Listen to AJAX Requests
+  browser.runtime.onMessage.addListener(request => {
+    let result = config.scrapeAjax(request);
+    if(result !== undefined && result !== null){
+      scrapedAjaxRows = result;
+      result.forEach((item) => {
+        scrapedAjaxRowDict[item.id] = item.dataValues;
+      });
+    }
+    loadTable();
```

Comment:
This is resulting in a ton of table reloads on pages which do a lot of requests, since it reloads on every request, which isn't ideal. Also, it requires the `scrapeAjax` function in the adapter to specify conditional logic in the scrape function for when to run or not, which feels a bit weird to me.

I think we should reconsider a better API surface here -- maybe the adapter specifies a function for which AJAX requests to scrape, like it specifies which URLs to activate on? Or could have a convention of returning `null` when you want to ignore a request and not reload everything?

Reply:
Actually maybe I'm overcomplicating this. Maybe it's as simple as moving this `loadTable` up a line into the if block. @TylerMillis @jessquaye thoughts?

Reply:
Yeah, you are right. Moving loadTable into the if block seems to solve this issue.